syntax = "proto3";
package hdf5_proto;

import "FileInfo.proto";
import "OpenFile.proto";
import "defs.proto";
import "Histogram.proto";
import "ImageData.proto";
import "SpectralProfile.proto";
// import "SpatialProfile.proto";
// import "Region.proto";

message Mask{
  repeated bool mask = 1;
  int32 width = 2;
  int32 height = 3;
}

service H5Readers{
  rpc CheckStatus(Empty) returns  (StatusResponse) {}
  rpc OpenFile(OpenFileRequest) returns (StatusResponse) {}
  rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
  rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}
  rpc GetImageDataStream (ImageDataRequest) returns (stream ImageDataResponse) {}

  rpc GetSpectralProfile (SpectralProfileReaderRequest) returns (SpectralProfileReaderResponse) {}
  // rpc GetSpectralProfileStream (SpectralProfileRequest) returns (stream SpectralProfileResponse) {}
  
  rpc GetHistogram (HistogramRequest) returns (HistogramResponse) {}
  rpc GetHistogramDist (HistogramDistRequest) returns (HistogramResponse) {}

  // rpc GetRegion (RegionDataRequest) returns (RegionDataResponse) {}

}

// service H5Services{
//   rpc CheckStatus(Empty) returns  (StatusResponse) {}
//   rpc OpenFile(OpenFileRequest) returns (OpenFileACK) {}
//   rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
//   rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}
//   rpc CreateRegion (SetRegion) returns (SetRegionAck) {}
//   rpc GetImageDataStream (ImageDataRequest) returns (stream ImageDataResponse) {}
//   rpc GetSpatialProfile (SetSpatialReq) returns (SpatialProfileData) {}
//   rpc GetSpectralProfile (SpectralProfileRequest) returns (SpectralProfileResponse) {} 
//   rpc GetHistogram (SetHistogramReq) returns (HistogramResponse) {}
// }